pkg/cluster/cluster.go

```diff
@@ -149,21 +148,24 @@ func (c *Cluster) setProcessName(procName string, args ...interface{}) {
 	}
 }
 
-func (c *Cluster) setStatus(status acidv1.PostgresStatus) {
-	// TODO: eventually switch to updateStatus() for kubernetes 1.11 and above
-	var (
-		err error
-		b   []byte
-	)
-	if b, err = json.Marshal(status); err != nil {
+// SetStatus of Postgres cluster
+// TODO: eventually switch to updateStatus() for kubernetes 1.11 and above
+func (c *Cluster) setStatus(status string) {
+	var pgStatus acidv1.PostgresStatus
+	pgStatus.PostgresClusterStatus = status
+
+	patch, err := json.Marshal(struct {
+		PgStatus interface{} `json:"status"`
+	}{&pgStatus})
+
+	if err != nil {
 		c.logger.Errorf("could not marshal status: %v", err)
 	}
 
-	patch := []byte(fmt.Sprintf(`{"status": %s}`, string(b)))
 	// we cannot do a full scale update here without fetching the previous manifest (as the resourceVersion may differ),
 	// however, we could do patch without it. In the future, once /status subresource is there (starting Kubernets 1.11)
 	// we should take advantage of it.
-	newspec, err := c.KubeClient.AcidV1ClientSet.AcidV1().Postgresqls(c.clusterNamespace()).Patch(c.Name, types.MergePatchType, patch)
+	newspec, err := c.KubeClient.AcidV1ClientSet.AcidV1().Postgresqls(c.clusterNamespace()).Patch(c.Name, types.MergePatchType, patch, "status")
```

Comment:
1. the `updateStatus()` in k8s codebase generally  does `PUT` instead of `PATCH` ([example](https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/extensions/v1beta1/deployment.go#L139)). Is this case the effect of patching `/status` always the same as completely replacing it with `PUT`? 
2.  actually our code generation should give us `updateStatus()` automatically. The code generation is adapted from etcd / Prometheus operators and the latter [definitely has this generated method](https://github.com/coreos/prometheus-operator/blob/0e22050ca3ebe6b7033f8c0f1cb1605e3abe63c5/pkg/client/versioned/typed/monitoring/v1/prometheus.go#L134)